features/step_definitions/aruba_steps.rb

```diff
@@ -0,0 +1,65 @@
+if Aruba::VERSION < '1.0.0'
+  require 'aruba/cucumber/core'
+end
+require 'aruba/generators/script_file'
+
+When(/^I run `([^`]*)` in interactive mode$/) do |cmd|
+  cmd = sanitize_text(cmd)
+  @interactive = run_command(cmd)
+  @processed_inputs = []
+  @processed_outputs = []
+end
+
+Then(/^I should see a prompt "(.+?)"$/) do |prompt|
+  @interactive.stop
+  current_output = sanitize_text(@interactive.output)
+  expect(current_output).to include_output_string(prompt)
+end
+
+When(/^I type "([^"]*)" and continue$/) do |input|
+  @interactive.start
+  @processed_inputs << input
+  @processed_inputs.each do |step_input|
+    type(unescape_text(step_input))
+  end
+end
+
+Then /^the last (\d) lines of (output|stderr|stdout) should contain: "(.+?)"$/ do |line_count, channel, expected_output|
+  output = case channel
+  when 'output'
+    all_output
+  when 'stderr'
+    all_stderr
+  when 'stdout'
+    all_stdout
+  end
```

Comment:
While reading this, I am asking myself: Are `all_output` & co Aruba methods? When they're called `all_*`, are there "partial *" counterparts?

Reply:
Yes they are aruba methods. `all_output` in this context means, the output of all commands that have been run in this scenario. The only "patial" output given by aruba therefore is the output of a specific command, which can be accessed  by the step  `Then(/^(?:the )?(output|stderr|stdout) from "([^"]*)" should( not)? contain( exactly)? "([^"]*)"$/)`. But there are no `partial_*` or similar counterparts

Reply:
Ok thanks.